tests/unit/util/main.c

```diff
@@ -167,6 +167,92 @@ ZTEST(util, test_UTIL_AND) {
 	zassert_equal(UTIL_AND(SEVEN, SEVEN), 7);
 }
 
+ZTEST(util, test_UTIL_ADD) {
+	zexpect_equal(UTIL_ADD(2048, 2047), 4095);
```

Comment:
Well, I did verify that they expand to actual literals, but you're right, I'll add some tests like
```.c
/* list with literal results */
#define Z_IS_LITERAL_4095    ,1

#define Z_IS_LITERAL(v, ...) UTIL_CHECK(~ TEST_IS_LITERAL_##v)
#define IS_LITERAL(v)        Z_IS_LITERAL(v)

ZTEST(util, test_UTIL_ADD) {
	zexpect_equal(UTIL_ADD(2048, 2047), 4095);
	/* rest of UTIL_ADD tests */

	zexpect_true(IS_LITERAL(UTIL_ADD(2048, 2047)));
}
```

Although, I have to mention that the values expand to literals by default. If that weren't the case, you'd receive a build-error message. (because the expansion from bits to integer would fail)

Regarding hex numbers, zephyr doesn't support hex numbers for the other macros, so I decided to also not support them. They pose too many issues... like 0x0001 and 0x1 are the same value, but you'd have to create every possible combination of leading zeroes that it isn't really worth it.